src/lksearch/MASTSearch.py

```diff
@@ -1115,53 +1128,64 @@ def _download_one(
 
         # We don't want to query cloud_uri if we don't have to
         # First check to see if we're not downloading on a cloud platform
-        # If not - cloud_uris should have already been queried - in that case
+        # If so - cloud_uris should have already been queried - in that case
         # check to see if a cloud_uri exists, if so we just pass that
 
-        download = True
+        skip_download = np.zeros_like(self.table.obs_id).astype(dtype=bool)
+        cloud_manifest = None
+        local_manifest = None
+        download_manifest = None
+
         if not conf.CHECK_CACHED_FILE_SIZES:
             # If this configuration parameter is set and the file exists
             # in the cache, we do not search for it
-            local_path = "/".join(
-                [
-                    config.get_cache_dir(),
-                    "mastDownload",
-                    row["obs_collection"],
-                    row["obs_id"],
-                    row["productFilename"],
-                ]
-            )
-            if os.path.isfile(local_path):
-                manifest = pd.DataFrame(
+            files_in_cache, local_paths = self._check_local_cache()
+
+            skip_download = skip_download | files_in_cache
+
+            if any(files_in_cache):
+                local_manifest = pd.DataFrame(
                     {
-                        "Local Path": [local_path],
+                        "Local Path": local_paths[files_in_cache],
                         "Status": ["UNKNOWN"],
-                        "Message": [None],
+                        "Message": [
+                            "File exists in local cache, has not been validated for integrity"
+                        ],
                         "URL": [None],
-                    }
+                    },
+                    index=self.table.index[files_in_cache],
                 )
-                return manifest
+
         if not conf.DOWNLOAD_CLOUD:
-            if pd.notna(row["cloud_uri"]):
-                download = False
-        if conf.DOWNLOAD_CLOUD or download:
-            manifest = Observations.download_products(
-                Table().from_pandas(row.to_frame(name=" ").transpose()),
+            cloud_uri_exists = self.table["cloud_uri"].notna()
+            skip_download = skip_download | cloud_uri_exists
+            if cloud_uri_exists.any():
+                cloud_manifest = pd.DataFrame(
+                    {
+                        "Local Path": [self.table.loc[cloud_uri_exists]["cloud_uri"]],
+                        "Status": ["COMPLETE"],
+                        "Message": ["Link to S3 bucket for remote read"],
+                        "URL": [None],
+                    },
+                    index=self.table.index[cloud_uri_exists],
+                )
+
+        if any(~skip_download):
+            download_manifest = Observations.download_products(
```

Comment:
Setting verbose=True for observations should show the progress. Does that happen here? Asking since we had to remove tqdm from the loop as it was done before.